framework_tvb/tvb/adapters/analyzers/wavelet_adapter.py

```diff
@@ -148,12 +148,6 @@ def configure(self, view_model):
         if view_model.sample_period is not None:
             algorithm.sample_period = view_model.sample_period
 
-        # TODO range form is not correctly populated, some work is still needed there
-        # if (view_model.frequencies is not None):
-        #     and 'lo' in frequencies_parameters
-        #         and 'hi' in frequencies_parameters and frequencies_parameters['hi'] != frequencies_parameters['lo']):
-        #     algorithm.frequencies = Range(**frequencies_parameters)
-
```

Comment:
Are you certain this part simply needs to be removed?
Are the values from UI being passed to the ViewModel and then into the algorithm at launch time?

Reply:
yes, at launch frequencies is in both ViewModel and algorithm at launch that is why I simply removed it.

Reply:
Please do a debug and choose other values than the defaults. I think it always uses the defaults if we do not explicitly do something. I think you should extend the fill_trait method on ContinuousWaveletTransformAdapterForm and explicitly prepare the Range object with values from RangeForm.